nixos/tests/alist.nix

```diff
@@ -0,0 +1,110 @@
+{ lib, pkgs, ... }:
+let
+  tls-cert = pkgs.runCommand "selfSignedCerts" { buildInputs = [ pkgs.openssl ]; } ''
+    openssl req \
+      -x509 -newkey rsa:4096 -sha256 -days 365 \
+      -nodes -out cert.pem -keyout key.pem \
+      -subj '/CN=alist' -addext "subjectAltName=DNS:alist"
+
+    mkdir -p $out
+    cp key.pem cert.pem $out
+  '';
+in
+{
+  name = "alist";
+
+  meta.maintainers = with lib.maintainers; [ moraxyc ];
+
+  nodes.machine =
+    { pkgs, ... }:
+    {
+      environment.systemPackages = with pkgs; [
+        curl
+        alist
+        sudo
+      ];
+      services.alist = {
+        enable = true;
+        settings = {
+          jwt_secret = {
+            _secret = "${pkgs.writeText "password" "jwtsecret"}";
+          };
+          scheme = {
+            https_port = 5245;
+            cert_file = "${tls-cert}/cert.pem";
+            key_file = "${tls-cert}/key.pem";
+          };
+        };
+      };
+    };
+
+  testScript = ''
+    import json
+
+    machine.wait_for_unit("alist.service")
+    machine.wait_for_open_port(5244)
+    machine.wait_for_open_port(5245)
+
+    # Ping
```

Comment:
Nit: these can different smaller tests can be instead placed under subtests, which makes logging cleaner

```python
with subtest("Ping"):
  ...
```